# Lab 8: Building a Table-Top Spectrometer
Shreya Parjan

5 November 2018

## Objective
The objective of this lab is to accurately measure the red and cyan wavelengths of hydrogen by building a spectrometer.

In [17]:
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats
import scipy.optimize
import math as m
%matplotlib inline

## Wavelength Calculations

In [18]:
m = 1 #unitless
thetaRed = np.radians(25) #rad
thetaCyan = np.radians(18.1) #rad
d = 8.3e-7 #m
y = 10e-2 #m
l = 30.5e-2 #m

phi = np.arctan(y/l)
print("phi = ", phi)
ldRed = 2*d*np.sin(thetaRed)*np.cos(phi)
ldCyan = 2*d*np.sin(thetaCyan)*np.cos(phi)
print("Red wavelength = ", ldRed)
print("Cyan wavelength = ",ldCyan)

phi =  0.3168244858682192
Red wavelength =  6.666300323890647e-07
Cyan wavelength =  4.900551090580305e-07


In [19]:
errorRed = ((ldRed - 656.3e-9)/(656.3e-9))*100
errorCyan = ((ldCyan - 486.1e-9)/(486.1e-9))*100
print("Percent error in red wavelength = ", errorRed, "%")
print("Percent error in cyan wavelength = ", errorCyan, "%")

Percent error in red wavelength =  1.573980251266906 %
Percent error in cyan wavelength =  0.8136410323041613 %


## Conclusion
Our measurements for the wavelengths of the cyan and red light only differed from the true values by about 0.8% and 1.6% respectively, indicating a high level of accuracy.